x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/reducer.ts

```diff
@@ -577,7 +577,8 @@ export function reducer(state: State, action: Action): State {
       const { jobConfig: config } = state;
       const { jobId } = state.form;
       // @ts-ignore
-      const formState = getFormStateFromJobConfig(config, false);
+      const formStateFromJobConfig = getFormStateFromJobConfig(config, false);
+      const formState = { ...formStateFromJobConfig, ...state.form };
```

Comment:
Can you maybe add a comment why this is necessary/what it is fixing. Just wondering, is there a chance that this needs to be a "deep" override since this version will completely replace any matching attributes on the most outer level and not override more granular nested settings.